### PR TITLE
Generate manifest from original. Better export.

### LINKED ISSUE #23: better next export support
currently `next-offline` only works with `next export` when the user has an `exportPathMap` `Next@6.0.0` now will generate the path map if the user doesn't declare one and `next-offline` should support that use case.

### LINKED ISSUE #45: Remove hard-coded assumptions
If I wanted to output the generated service worker to any other place (e.g. `.next/static` or `static` - so they are automatically statically served) using `workboxOpts.swDest` then everything else breaks 😬

### LINKED ISSUE #49: Use workbox precache manifest output
While digging in I found out that `workbox-webpack-plugin` limits all of the "transform" options to the `globPattern` files only. I.e. don't run `modifyUrlPrefix` on webpack-generated assets, for example. See whole explanation [here](https://github.com/GoogleChrome/workbox/issues/1591#issuecomment-415727436).

I then found out this will be available in `workbox-webpack-plugin` v4: https://github.com/GoogleChrome/workbox/issues/1591

We can still do all of that work manually:

1. Read contents of `precache-manifest.*.js`
2. Mimic `modifyUrlPrefix` according to the existing `exportPathMap`
3. Remove revision from hashed files (maybe all)
4. Ignore unserved files (e.g. `build-manifest.json`)

I'd be ok to wait for v4, given the current version is working fine. Reading from the generated precache manifest is just a more accurate process.

### PR DESCRIPTION
**Pull request by joaovieira**
*Luckily* the Next 7 builds everything inside `/static` so the previous version just works 🎉 

This PR refactors the logic as discussed in #49 to read the workbox-generated precache manifest instead of generating one from scratch. This should be more reliable moving forward, in case Next.js updates their paths.

On top of that I've made the export functionality using the new `exportPathMap` options which simplifies massively. #23

Finally, I've removed the hard-coded service-worker.js to use `workboxOpts.swDest` everywhere. #45 

**This is not compatible with next@6 so it requires a major bump.**

Fixes #49.
Fixes #23.
Fixes #45.

### CODE REVIEW

**Comment by joaovieira** [comment]
I actually thought about mentioning it. But it really just started with the #49, then I was just carried away with the rest.. 😆But do let me know if you prefer things in any other way!

No rush for the review. I forgot to change the next requirements in package.json. Doing it now.

**Comment by hanford** [comment]
ahh, @joaovieira will we need to make make the `next` peerDep explicitly 7.0.0+? I'm happy to make this a breaking change and putting this in a 3.0.0 release!

major improvement to the codebase IMO 🍾🍻

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "printWidth": 100,
+  "trailingComma": "all",
+  "singleQuote": true
```
Comment (joaovieira) [inline .prettierrc.json]: put some defaults that I use personally. the only difference I think was the `trailingComma` option. what do you prefer?

File: package.json [hunk 2]
```diff
@@ -36,21 +39,20 @@
...
     "copy-webpack-plugin": "~4.5.2",
-    "fs-extra": "~5.0.0",
-    "glob": "~7.0.0",
-    "minimist": "1.2.0",
-    "webpack": "~3.11.0",
-    "workbox-build": "^3.5.0",
-    "workbox-webpack-plugin": "^3.5.0"
+    "fs-extra": "~7.0.0",
+    "glob": "~7.1.3",
+    "webpack": "^4.19.1",
```
Comment (joaovieira) [inline package.json]: still required by workbox plugin even though it's not a peer dependency 😩

File: .prettierrc.json (hunk 1 above)
Comment (hanford) [reply .prettierrc.json]: Trailing makes sense

**Comment by hanford** [comment]
Thanks again ❤️

**Comment by hanford** [comment]
Live in 3.0.0 🔥🎉

**Pull request merged**